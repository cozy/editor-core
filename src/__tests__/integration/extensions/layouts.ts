import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  getDocFromElement,
  editable,
  insertBlockMenuItem,
  changeSelectedNodeLayout,
} from '../_helpers';
import {
  mountEditor,
  goToEditorTestingExample,
} from '../../__helpers/testing-example-helpers';

import commonMessages from '../../../messages';

[
  commonMessages.layoutFixedWidth,
  commonMessages.layoutWide,
  commonMessages.layoutFullWidth,
].forEach(layoutMessages => {
  const layoutName = layoutMessages.id.split('.').pop();
  BrowserTestCase(
    `layouts.ts: Extension: ${layoutName} layout`,
    { skip: ['edge'] },
    async (client: any, testName: string) => {
      const page = await goToEditorTestingExample(client);
      await mountEditor(page, {
        appearance: 'full-page',
        allowExtension: {
          allowBreakout: true,
        },
      });

      await insertBlockMenuItem(page, 'Block macro (EH)');
      await changeSelectedNodeLayout(page, layoutMessages.defaultMessage);

      const doc = await page.$eval(editable, getDocFromElement);
      expect(doc).toMatchCustomDocSnapshot(testName);
    },
  );
});
