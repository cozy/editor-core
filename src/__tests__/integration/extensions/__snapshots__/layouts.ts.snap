// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts.ts: Extension: layoutFixedWidth layout 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "extensionKey": "block-eh",
        "extensionType": "com.atlassian.confluence.macro.core",
        "layout": "default",
        "localId": "testId",
        "parameters": Object {
          "macroMetadata": Object {
            "placeholder": Array [
              Object {
                "data": Object {
                  "url": "",
                },
                "type": "icon",
              },
            ],
          },
          "macroParams": Object {},
        },
        "text": "Block extension demo",
      },
      "type": "extension",
    },
  ],
  "type": "doc",
}
`;

exports[`layouts.ts: Extension: layoutFullWidth layout 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "extensionKey": "block-eh",
        "extensionType": "com.atlassian.confluence.macro.core",
        "layout": "full-width",
        "localId": "testId",
        "parameters": Object {
          "macroMetadata": Object {
            "placeholder": Array [
              Object {
                "data": Object {
                  "url": "",
                },
                "type": "icon",
              },
            ],
          },
          "macroParams": Object {},
        },
        "text": "Block extension demo",
      },
      "type": "extension",
    },
  ],
  "type": "doc",
}
`;

exports[`layouts.ts: Extension: layoutWide layout 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "extensionKey": "block-eh",
        "extensionType": "com.atlassian.confluence.macro.core",
        "layout": "wide",
        "localId": "testId",
        "parameters": Object {
          "macroMetadata": Object {
            "placeholder": Array [
              Object {
                "data": Object {
                  "url": "",
                },
                "type": "icon",
              },
            ],
          },
          "macroParams": Object {},
        },
        "text": "Block extension demo",
      },
      "type": "extension",
    },
  ],
  "type": "doc",
}
`;
